src/azure-cli/azure/cli/command_modules/vm/disk_encryption.py

```diff
@@ -95,32 +95,29 @@ def updateVmEncryptionSetting(cmd, vm, resource_group_name, vm_name, encryption_
 
 def updateVmssEncryptionSetting(cmd, vmss, resource_group_name, vmss_name, encryption_identity):
     from azure.cli.core.azclierror import ArgumentUsageError
-    if vmss.identity is None or vmss.identity.user_assigned_identities is None or encryption_identity.lower() not in \
-            (k.lower() for k in vmss.identity.user_assigned_identities.keys()):
+    if encryption_identity.lower() not in (k.lower() for k in vmss.get('identity', {}).get('userAssignedIdentities', {}).keys()):
         raise ArgumentUsageError("Encryption Identity should be an ARM Resource ID of one of the "
                                  "user assigned identities associated to the resource")
 
-    VirtualMachineProfile, SecurityProfile, EncryptionIdentity \
-        = cmd.get_models('VirtualMachineProfile', 'SecurityProfile', 'EncryptionIdentity')
     updateVmss = False
-
-    if vmss.virtual_machine_profile is None:
-        vmss.virtual_machine_profile = VirtualMachineProfile()
-    if vmss.virtual_machine_profile.security_profile is None:
-        vmss.virtual_machine_profile.security_profile = SecurityProfile()
-    if vmss.virtual_machine_profile.security_profile.encryption_identity is None:
-        vmss.virtual_machine_profile.security_profile.encryption_identity = EncryptionIdentity()
-    if vmss.virtual_machine_profile.security_profile.encryption_identity.user_assigned_identity_resource_id\
-        is None or vmss.virtual_machine_profile.security_profile.encryption_identity\
-            .user_assigned_identity_resource_id.lower() != encryption_identity:
-        vmss.virtual_machine_profile.security_profile.encryption_identity.user_assigned_identity_resource_id \
-            = encryption_identity
+    if vmss['properties']['virtualMachineProfile'].get('securityProfile', {}).get('encryptionIdentity', {}).\
+            get('userAssignedIdentityResourceId', '').lower() != encryption_identity.lower():
         updateVmss = True
 
     if updateVmss:
-        compute_client = _compute_client_factory(cmd.cli_ctx)
-        updateEncryptionIdentity \
-            = compute_client.virtual_machine_scale_sets.begin_create_or_update(resource_group_name, vmss_name, vmss)
+        from .aaz.latest.vmss import Patch
+        virtual_machine_profile = {
+            'securityProfile': {
+                'encryptionIdentity': {
+                    'userAssignedIdentityResourceId': encryption_identity
+                }
+            }
+        }
+        updateEncryptionIdentity = Patch(cli_ctx=cmd.cli_ctx)(command_args={
```

Comment:
May I ask why we need to change the http request from PUT method to PATCH method when updating the resource?